### PR TITLE
test: Remove appstream-util

### PR DESCRIPTION
**Pull request by ghost**
Flathub prefers appstreamcli now.

### CODE REVIEW

**Comment by realmazharhussain** [comment]
https://github.com/gdm-settings/gdm-settings/pull/237#issuecomment-2030079511

> I think you can drop appstream-glib now. Flathub prefers appstreamcli.
> 
> https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/#validation

GDM Settings is primarily a system application, and because of that we cannot be sure that `appstream-glib` is not installed on someone's system even if we don't depend on it. In my case, I have Flatseal installed on my system, and it depends on `appstream-gib`. So, currently, `translate="no"` properties don't work on my system.

Also, flatpak is not the preferred way of installing GDM Settings, as it is a system application and works better without the sandbox.

File: data/gdm-settings.metainfo.xml [hunk 1]
```diff
@@ -84,15 +84,15 @@
   <releases>
     <release version="4.3" type="stable" date="2024-02-25">
       <url>https://github.com/gdm-settings/gdm-settings/releases/tag/v4.3</url>
-      <description translatable="no" translate="no">
+      <description translate="no">
```
Comment (realmazharhussain) [inline data/gdm-settings.metainfo.xml:87]: This won't work. See https://github.com/gdm-settings/gdm-settings/pull/238#issuecomment-2031015790 for more info.

File: io.github.realmazharhussain.GdmSettings.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ app-id: io.github.realmazharhussain.GdmSettings
 command: gdm-settings
 sdk: org.gnome.Sdk
 runtime: org.gnome.Platform
-runtime-version: "46"
+runtime-version: "master"
```
Comment (realmazharhussain) [inline io.github.realmazharhussain.GdmSettings.yaml:5]: GDM Settings is not a GNOME offering. GNOME apps need to use the nightly runtime because they are supposed to be developed and released alongside it. However, third party apps should depend on a versioned runtime.

Until GNOME 47 beta (or maybe even stable) is released, GDM Settings' main branch will use GNOME 46 runtime.

File: io.github.realmazharhussain.GdmSettings.yaml [hunk 3]
```diff
@@ -39,6 +39,7 @@ modules:
   - name: gdm-settings
     builddir: true
     buildsystem: meson
+    run-tests: true
```
Comment (realmazharhussain) [inline io.github.realmazharhussain.GdmSettings.yaml:42]: This should be a separate commit. This change is not related to "Remove appstream-util".

**Review by realmazharhussain — CHANGES_REQUESTED** [review]

File: io.github.realmazharhussain.GdmSettings.yaml (hunk 2 above)
Comment (yakushabb) [reply io.github.realmazharhussain.GdmSettings.yaml:5]: I used it to build the project, it was development intended.

File: io.github.realmazharhussain.GdmSettings.yaml (hunk 3 above)
Comment (yakushabb) [reply io.github.realmazharhussain.GdmSettings.yaml:42]: It was development intended to test appdata files.

File: data/meson.build [hunk 4]
```diff
@@ -9,18 +9,9 @@ appstream_file = i18n.merge_file(
 appstreamcli = find_program('appstreamcli', required: false)
 if appstreamcli.found()
   test(
-    'Validate appstream file (with appstreamcli)',
+    'Validate appstream file',
     appstreamcli,
-    args: ['validate', '--no-net', appstream_file]
```
Comment (realmazharhussain) [inline data/meson.build:14]: It seems like my other comments were demotivating. Sorry!

I was shedding off unrelated stuff.

This part of the PR is actually useful.

Thank you for submitting this PR. Would you like to reopen it?

**Comment by realmazharhussain** [comment]
22.4
